server/build_event_protocol/event_parser/event_parser.go

```diff
@@ -175,188 +191,207 @@ func (sep *StreamingEventParser) ParseEvent(event *inpb.InvocationEvent) {
 			if wfc == nil {
 				return
 			}
-			sep.workflowConfigurations = append(sep.workflowConfigurations, wfc)
+			sep.fillInvocationFromWorkflowConfigured(wfc)
 		}
 	}
 }
 
-func (sep *StreamingEventParser) FillInvocation(invocation *inpb.Invocation) {
-	invocation.Command = sep.command
-	invocation.Pattern = sep.pattern
-	invocation.Event = sep.events
-	invocation.Success = sep.success
-	invocation.ActionCount = sep.actionCount
-
-	// Fill invocation in a deterministic order:
-	// - Environment variables
-	// - Workspace status
-	// - Build metadata
-
-	for _, commandLine := range sep.structuredCommandLines {
-		fillInvocationFromStructuredCommandLine(commandLine, invocation)
-	}
-	for _, workspaceStatus := range sep.workspaceStatuses {
-		fillInvocationFromWorkspaceStatus(workspaceStatus, invocation)
-	}
-	for _, buildMetadatum := range sep.buildMetadata {
-		fillInvocationFromBuildMetadata(buildMetadatum, invocation)
-	}
-	for _, workflowConfigured := range sep.workflowConfigurations {
-		fillInvocationFromWorkflowConfigured(workflowConfigured, invocation)
-	}
-
-	buildDuration := time.Duration(int64(0))
-	if sep.endTime != nil && sep.startTime != nil {
-		buildDuration = sep.endTime.Sub(*sep.startTime)
-	}
-	invocation.DurationUsec = buildDuration.Microseconds()
-	if sep.terminalWriter != nil {
-		// TODO(siggisim): Do this rendering once on write, rather than on every read.
-		invocation.ConsoleBuffer = string(sep.terminalWriter.Render())
-	}
-}
-
-func fillInvocationFromStructuredCommandLine(commandLine *command_line.CommandLine, invocation *inpb.Invocation) {
+func (sep *StreamingEventParser) fillInvocationFromStructuredCommandLine(commandLine *command_line.CommandLine) {
 	envVarMap := parseEnv(commandLine)
-	if commandLine != nil {
-		invocation.StructuredCommandLine = append(invocation.StructuredCommandLine, commandLine)
-	}
 	if user, ok := envVarMap["USER"]; ok && user != "" {
-		invocation.User = user
+		sep.setUser(user, envPriority)
```

Comment:
It feels odd to me to use the const so many times, maybe i'm just weird. wdyt about doing:
```
priority := envPriority

sep.setUser(user, priority)
...
```

here and below?

Reply:
That does feel slightly cleaner - done